### PR TITLE
Add CoAP examples

### PR DESCRIPTION
**Pull request by alashkin**
Hi, @mmikulicic!

This is last PR for CoAP minimal.
Added two standard protocol samples: coap_client & coap_server.

Next PRs (if any) will be some kind of additions (depends on discussion on document I'm going to write).

### CODE REVIEW

File: examples/coap_client/coap_client.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+static void coap_handler(struct ns_connection *nc, int ev, void *p) {
+  switch (ev) {
+    case NS_CONNECT: {
+      struct ns_coap_message cm;
+      uint32_t res;
+
+      memset(&cm, 0, sizeof(cm));
+      cm.msg_id = 1;
+      cm.msg_type = NS_COAP_MSG_CON;
+      fprintf(stdout, "Sending CON...\n");
```
Comment (mmikulicic) [inline examples/coap_client/coap_client.c]: `s/fprintf(stdout/printf(`

File: examples/coap_client/coap_client.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * Copyright (c) 2015 Cesanta Software Limited
+ * All rights reserved
+ *
+ * This program sends CoAP CON-message to server (coap.me by default) 
+ * and waits for answer.
+ */
+
+#include "fossa.h"
+
+static int g_time_to_exit = 0;
```
Comment (mmikulicic) [inline examples/coap_client/coap_client.c]: for consistency let's use the `s_` prefix (for static, not for global).

File: examples/coap_server/coap_server.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * All rights reserved
+ *
+ * This program listens on 5683 for CoAP messages, 
+ * sends ACK is nessesary and dump everything received.
+ * It is possible to use ../coap_client to send message.
+ */
+
+#include "fossa.h"
+
+static char* g_default_address = "udp://5683";
```
Comment (mmikulicic) [inline examples/coap_server/coap_server.c]: `udp://:5683` is the documented format, this without colon works but should be eventually removed and it shouldn't be present in our examples.

File: examples/coap_client/coap_client.c (hunk 1 above)
Comment (alashkin) [inline examples/coap_client/coap_client.c]: Why?
fprintf flushes stdout unlike printf

**Comment by mmikulicic** [comment]
LGTM, modulo minor comments

File: examples/coap_client/coap_client.c (hunk 1 above)
Comment (mmikulicic) [inline examples/coap_client/coap_client.c]: On which platform?

AFAIK both behave the same, and flush the output on newline.

Comment (alashkin) [inline examples/coap_client/coap_client.c]: Actually, I forgot what I added \n. 
So, you are right.

File: examples/coap_client/coap_client.c (hunk 2 above)
Comment (alashkin) [inline examples/coap_client/coap_client.c]: Done.

File: examples/coap_server/coap_server.c (hunk 3 above)
Comment (alashkin) [inline examples/coap_server/coap_server.c]: Done.

**Comment by alashkin** [comment]
PTAL

**Pull request merged**